flytekit/remote/remote.py

```diff
@@ -603,7 +603,8 @@ def _execute(
             disable_all = None
 
         with self.remote_context() as ctx:
-            literal_inputs = TypeEngine.dict_to_literal_map(ctx, inputs)
+            python_types = TypeEngine.guess_python_types(entity.interface.inputs)
```

Comment:
this isn't quite what we talked about right?  can we add `guessed_python_interface` to FlyteTask, FlyteLaunchPlan and FlyteWorkflow?  And then in the cases where they are fetched from a local `@task` or `@workflow` the guessed interface is just assigned from the known local python interface.